### DISCUSSION TITLE
live usb setup

### DISCUSSION THREAD

**Discussion by scriptkitten24** [linux-surface/linux-surface#576] [Q&A]
Hi! Any recommendations for creating a live USB instead of a full install?  I have a Surface Book 2 and [already had] an Ubuntu Studio live USB that I have tried. The built-in keyboard and mouse are not working (without modifications thus far). I thought I'd check to see if anyone has attempted this. Any Linux distros that may be a better starting place? It's my company machine, so I cannot modify the hard drive. It would be awesome to use Kali Linux or other live boot toolkits as well. Curious if anyone has attempted anything like this on a Surface.

**Comment by qzed** [comment]
Can't really help you with your actual question (i.e. creating a live USB), but I can maybe provide some related info that may help you and others:

Anything with a v5.13 or later kernel should work out-of-the-box for most devices (meaning anything that's currently available excluding AMD variants) with maybe some caveats. For some devices (Surface Book 2, Surface Pros, but not Surface Laptops) you can also go with a v5.12 kernel, however, the Surface devices that work with v5.12 should also have basic functionality with earlier kernels (minus battery status and some quirks). Specifically the keyboard should work on those (only the Surface Laptops and Surface Book 3 need a special driver for that). On earlier kernels there is, however, an issue that pretty much renders the Surface Book 2 unusable due to ACPI causing a kernel thread to spin, heating up the processor and consuming battery needlessly, so you still want to go with v5.12+ for that.

Unfortunately I don't think there are any ISOs out yet with those newer kernel versions besides Arch (and derivatives). So you're correct that you'll have to create your own. Another thing to note is that upstream kernels don't yet include some WiFi fixes (for anything up to, excluding, Surface Laptop 3, Surface Book 3, and Surface Pro 7) and touch support (for anything from, including, SB1, SP4 and later). Again something that you may need a `-surface` kernel for.

If your only concern is about not modifying the SSD though, you could consider installing to a SD card or a USB stick/external SSD or something like that. That will probably provide you with the best experience at the moment.
